### PR TITLE
[FIX] mail_tracking: use recipient_address to search emails

### LINKED ISSUE #803: [14.0] mail_tracking: button 'Tracking emails' in partner is showing a different count that shown when you click on it
The mail_tracking in v13 is searching the emails as follows:

https://github.com/OCA/social/blob/78daea28e5763cb132c77fcc986408680dd4e84f/mail_tracking/models/mail_tracking_email.py#L203-L214

It's using a regex to catch cases when the recipient is `<(emailexample@.example.com)>` and if regex is not matched, it's using the email like comes in the record, but in v14.0 is using the function `email_split` and it's only considering when exactly matching and excluding the emails as `<(emailexample@.example.com)>`

https://github.com/OCA/social/blob/147eda96a56c199602c619b5bac3a9911605e918/mail_tracking/models/mail_tracking_email.py#L204-L210

## Module
mail_tracking

## Describe the bug

The mail_tracking in v13 is searching the emails as follows:

https://github.com/OCA/social/blob/78daea28e5763cb132c77fcc986408680dd4e84f/mail_tracking/models/mail_tracking_email.py#L203-L214

It's using a regex to catch cases when the recipient is `<(emailexample@.example.com)>` and if regex is not matched, it's using the email like comes in the record, but in v14.0 is using the function `email_split` and it's only considering the when exactly matching and excluding the emails as `<(emailexample@.example.com)>`

https://github.com/OCA/social/blob/147eda96a56c199602c619b5bac3a9911605e918/mail_tracking/models/mail_tracking_email.py#L204-L210

## To Reproduce
14.0

1. Go to any partner with mails sent.
2. See the emails shown in button `Tracking emails` 
3. Click on button `Tracking emails` 
4. See that is likely it doesn't show the emails correctly

**Expected behavior**
The expected behavior is user sees the same count of emails in `Tracking emails` that when you click on it.

### PR DESCRIPTION
**Pull request by fernandahf**
Description of the issue/feature this PR addresses:

This commit creates a migration script to fix all emails computed incorrectly with old version of `email_split` 

Also, we are using the field `recipient_address` in the filter of the search view
of `mail.tracking.email` in order to use the same field used in computed
method:

https://github.com/OCA/social/blob/147eda96a56c199602c619b5bac3a9911605e918/mail_tracking/models/res_partner.py#L27

Current behavior before PR:
Inconsistencies between the counter field in partners to show the number of emails tracked and the shown when clicking on itself.

Desired behavior after PR is merged:
Show correctly the same number of emails tracked in the field in partners than the shown when clicking on itself.

Fix https://github.com/OCA/social/issues/803

-- I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fernandahf** [comment]
@moylop260 

Could you review this, please?

**Comment by moylop260** [comment]
Could you check the unittest failing in the CI, please?

Also, Could you tag to maintainer/authors of the module, please?

**Comment by fernandahf** [comment]
@pedrobaeza 

Could review this, please?

File: mail_tracking/__manifest__.py
Comment (pedrobaeza) [inline mail_tracking/__manifest__.py]: Don't change manually the version. The bot will do it.

File: mail_tracking/migrations/14.0.2.0.1/README.rst
Comment (pedrobaeza) [inline mail_tracking/migrations/14.0.2.0.1/README.rst]: This file should go out

File: mail_tracking/migrations/14.0.2.0.1/post-migration.py
Comment (pedrobaeza) [inline mail_tracking/migrations/14.0.2.0.1/post-migration.py:9]: Why do you need this?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
There's a method for doing the email split stuff, and if it's not enough, you should fix such method in Odoo, not to switch again to manual regular expressions that can fail in thousand ways. I don't approve this change.

File: mail_tracking/migrations/14.0.2.0.1/README.rst
Comment (fernandahf) [reply mail_tracking/migrations/14.0.2.0.1/README.rst]: Done

File: mail_tracking/__manifest__.py
Comment (fernandahf) [reply mail_tracking/__manifest__.py]: Done

**Comment by fernandahf** [comment]
@pedrobaeza 

In the last version of Odoo v14.0 error with the `email_split` function is not reproduced, I still think the migration to fix `recipient_address` is needed, what do you think?

**Comment by pedrobaeza** [comment]
I don't see the reason to merge this PR, as you are limiting the search to only the address, and the migration script seems a problem on your DB, not a general one.

**Comment by fernandahf** [comment]
@pedrobaeza 

I think it's an error to use a field to compute a counter and use another field to search them?

**Comment by pedrobaeza** [comment]
OK, I get your point. I agree, but you should remove the migration script.

**Comment by fernandahf** [comment]
Done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**
